From 2fc787e67a83a15fd72536a503342e4ab8a1784c Mon Sep 17 00:00:00 2001
From: Ira Cooper <ira@samba.org>
Date: Thu, 3 Mar 2016 13:47:32 -0500
Subject: [PATCH 1069/1096] source3: Honor the core soft limit of the OS.

We should honor the soft limits set by the operating system.

In any case, 16M doesn't make a useful coredump for modern
Samba.

Signed-off-by: Ira Cooper <ira@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>

Autobuild-User(master): Jeremy Allison <jra@samba.org>
Autobuild-Date(master): Sat Mar  5 00:39:48 CET 2016 on sn-devel-144

(cherry picked from commit 58d3462bc58290d8eb5e554c6c59cf6b73ccf58a)

BUG: https://bugzilla.samba.org/show_bug.cgi?id=11907
(cherry picked from commit 79cbf029f1846f82e3df67899c991ee0c57d1d44)
---
 source3/lib/dumpcore.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/source3/lib/dumpcore.c b/source3/lib/dumpcore.c
index 487ccec..c72aa88 100644
--- a/source3/lib/dumpcore.c
+++ b/source3/lib/dumpcore.c
@@ -251,21 +251,6 @@ void dump_core_setup(const char *progname, const char *log_file)
 		goto out;
 	}
 
-
-#ifdef HAVE_GETRLIMIT
-#ifdef RLIMIT_CORE
-	{
-		struct rlimit rlp;
-		getrlimit(RLIMIT_CORE, &rlp);
-		rlp.rlim_cur = MAX(16*1024*1024,rlp.rlim_cur);
-		setrlimit(RLIMIT_CORE, &rlp);
-		getrlimit(RLIMIT_CORE, &rlp);
-		DEBUG(3,("Maximum core file size limits now %d(soft) %d(hard)\n",
-			 (int)rlp.rlim_cur,(int)rlp.rlim_max));
-	}
-#endif
-#endif
-
 	/* FIXME: if we have a core-plus-pid facility, configurably set
 	 * this up here.
 	 */
-- 
2.9.0

